allensdk/brain_observatory/behavior/session_apis/data_io/behavior_ophys_lims_api.py

```diff
@@ -106,20 +106,94 @@ def get_nwb_filepath(self) -> str:
+                LEFT JOIN well_known_files wkf ON wkf.attachable_id = oe.ophys_session_id
+                JOIN well_known_file_types wkft ON wkf.well_known_file_type_id = wkft.id
                 WHERE wkf.attachable_type = 'OphysSession'
-                AND wkft.name = 'EyeTracking Ellipses'
-                AND oe.id = {};
-                """.format(self.get_ophys_experiment_id())
+                    AND wkft.name = 'EyeTracking Ellipses'
+                    AND oe.id = {self.get_ophys_experiment_id()};
+                """
+        return safe_system_path(self.lims_db.fetchone(query, strict=True))
+
+    @memoize
+    def get_eye_gaze_mapping_file_path(self) -> str:
+        """Get the filepath of the eye gaze mapping file (*.h5) associated with the
+        ophys experiment"""
+        query = f"""
+                SELECT wkf.storage_directory || wkf.filename AS eye_tracking_file
+                FROM ophys_experiments oe
+                LEFT JOIN well_known_files wkf ON wkf.attachable_id = oe.ophys_session_id
+                JOIN well_known_file_types wkft ON wkf.well_known_file_type_id = wkft.id
+                WHERE wkf.attachable_type = 'OphysSession'
+                    AND wkft.name = 'EyeDlcScreenMapping'
+                    AND oe.id = {self.get_ophys_experiment_id()};
+                """
         return safe_system_path(self.lims_db.fetchone(query, strict=True))
 
+    @memoize
+    def get_eye_tracking_rig_geometry(self) -> Optional[dict]:
+        """Get the eye tracking rig geometry metadata"""
+        ophys_experiment_id = self.get_ophys_experiment_id()
+
+        query = f'''
+            SELECT oec.*, oect.name as config_type, equipment.name as equipment_name
+            FROM ophys_sessions os
+            JOIN observatory_experiment_configs oec ON oec.equipment_id = os.equipment_id
+            JOIN observatory_experiment_config_types oect ON oect.id = oec.observatory_experiment_config_type_id
+            JOIN ophys_experiments oe ON oe.ophys_session_id = os.id
+            JOIN equipment ON equipment.id = oec.equipment_id
+            WHERE oe.id = {ophys_experiment_id} AND oect.name IN ('eye camera position', 'led position', 
+                'screen position')
+        '''
+        api = (credential_injector(LIMS_DB_CREDENTIAL_MAP)
+               (PostgresQueryMixin)())
+        # Get the raw data
+        rig_geometry = pd.read_sql(query, api.get_connection())
+
+        if rig_geometry.empty:
+            # There is no rig geometry for this experiment
+            return None
+
+        # Map the config types to new names
+        rig_geometry_config_type_map = {
+            'eye camera position': 'camera',
+            'screen position': 'monitor',
+            'led position': 'led'
+        }
+        rig_geometry['config_type'] = rig_geometry['config_type'].map(rig_geometry_config_type_map)
+
+        # There are multiple entries for a given config type. Select the most recent one
```

Comment:
I think rig geometry may need to be looked up based on `date_of_acquisition`.

For example, if an experiment was acquired 2020-05-15 and there was a rig geometry for 2020-01-12 as well as for 2020-07-25, then the correct geometry to use would be the 2020-01-12. Since the most recent geometry would have been set after the experiment was acquired.

This is how the LIMS strategy does it when creating the input.json used for populating the BehaviorOphysJsonApi:
http://stash.corp.alleninstitute.org/projects/TECH/repos/lims/browse/app/strategies/behavior_ophys_write_nwb_strategy.rb#54
http://stash.corp.alleninstitute.org/projects/TECH/repos/lims/browse/app/models/observatory_experiment_config.rb

Reply:
Good catch @njmei, I will correct this.